### DISCUSSION TITLE
Ignore all other unused members

### DISCUSSION THREAD

**Discussion by shantiNaser** [riok/mapperly#790] [Q&A]
Hello, I'm just wondering if there's a way supported by Mapperly that help Ignore all unused members?

**Comment by latonz** [comment]
What do you mean by unused? You can:
* [Ignore a single property/field](https://mapperly.riok.app/docs/configuration/mapper/#ignore-properties--fields)
* [Ignore the not mapped warnings with editorconfig](https://mapperly.riok.app/docs/configuration/analyzer-diagnostics/#editorconfig), see also [this](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/code-style-rule-options).
* Not yet implemented: Ignore all unmapped source or target, see https://github.com/riok/mapperly/issues/687

**Reply by shantiNaser** [reply]
@latonz 
I mean I have a DTO class that contains different properties - ( about 100 ) 
I mean by unused - it declared hard coded ( manually by a method ) not by map to a source data 

the solution to ignore it is by 
[MapperIgnoreTarget(nameof(ReportData.Value1))]

and so on 

I'm searching for a way to ignore all these members without manually adding the previous line for all properties

**Reply by latonz** [reply]
You only want to map properties explicitly mapped with `MapProperty`? IMO then it is probably easier to implement the mapping by hand. If you want to ignore all source **or** target members, you may need to wait until https://github.com/riok/mapperly/issues/687 lands.

**Reply by shantiNaser** [reply]
@latonz , I will try to explain my issue 
assume I have this class 

```csharp
 public class ReportData
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Number { get; set; }
        .......................
      }
```
I have created this Mapper Class
```csharp
public partial class ReportMapper
    {
        public ReportData MapToDtoModel(DomainData createParams)
        {
            var model = MapToDtoModelInternal(createParams);

           model.number = createParams.EmployeeNumber
           ................................

            return model;
        }
        
        [MapperIgnoreTarget(nameof(ReportData.Name))]
        private partial ReportData MapToDtoModelInternal(DomainData  createParams);
   }
```
        
in different area of a code       


```csharp
var result = GetReportData()

foreach (var p in result)
{
    var mappedModel = this.Mapper.MapToDtoModel(p);

    mappedModel.Name = GetFullName(name)
   // the other propriety filed here so i need to manually add a line to ignore it from target
}

return mappedModels;
}
```
 
Now I mean is there a way that i can ignore all name and other propriety that declared like this ?

**Reply by latonz** [reply]
I don't understand your use case... Why do you plan to use Mapperly while planing to ignore all properties at the same time? This way using Mapperly wouldn't bring any benefit (if all properties are ignored, it doesn't map anything?) Sorry if I don't understand.

**Reply by latonz** [reply]
Closing this as no further information was provided. Feel free to reopen with more information.
